satpy/readers/eum_l2_grib.py

```diff
@@ -75,10 +74,8 @@ def start_time(self):
     @property
     def end_time(self):
         """Return the sensing end time."""
-        if self.sensor == "seviri":
-            return self.start_time + timedelta(minutes=REPEAT_CYCLE_DURATION)
-        elif self.sensor == "fci":
-            return self.filename_info["end_time"]
+        delta = REPEAT_CYCLE_DURATION_RSS if self._ssp_lon == 9.5 else REPEAT_CYCLE_DURATION
+        return self.start_time + delta
```

Comment:
My comment only considered SEVIRI, since for SEVIRI we have a nominal repeat cycle duration of 15 minutes for FES and 5 minutes for RSS. For FCI, however, it's 10minutes. I also think we need to convert the imported integers to datetime.timedela objects? So we should have something like this instead:
```
if self.sensor == "seviri":
        delta = REPEAT_CYCLE_DURATION_RSS if self._ssp_lon == 9.5 else REPEAT_CYCLE_DURATION
        return self.start_time + timedelta(minutes=delta)
elif self.sensor == "fci":
        return self.filename_info["end_time"]
 ```